src/parse_file.py

```diff
@@ -204,6 +204,40 @@ def parse_ascii_directive(line: str, z: bool) -> bytes:
     return b"".join(ret)
 
 
+def parse_incbin(args: List[str], options: Options) -> Optional[bytes]:
+    if not options.incbin_dirs:
+        return None
+
+    try:
+        # TODO: Reuse ASCII string parser, instead of just stripping quotes
+        filename = args[0].strip("'\"")
+        offset = int(args[1].strip(), 0)
+        size = int(args[2].strip(), 0)
+    except ValueError:
+        raise DecompFailure(f"Could not parse asm_data .incbin directive: {args}")
+
+    for incbin_dir in options.incbin_dirs:
+        full_path = incbin_dir / filename
+        try:
+            with full_path.open("rb") as f:
+                f.seek(offset)
+                data = f.read(size)
+        except OSError:
+            continue
+
+        if len(data) != size:
+            print(
+                f"/* Unable to read {size} bytes from {full_path} at {offset:#x} (got {len(data)} bytes) */"
+            )
+            return None
+        return data
+
+    print(
+        f"/* Unable to find {filename} in any of {len(options.incbin_dirs)} search paths */"
+    )
```

Comment:
This warning could be useful even if not passing any incbin_dirs at all, together with a note about which command line flag allows that. Though we might need to deduplicate warnings so it doesn't get spammy.

Reply:
I deduped using a global variable, do you think that's an OK compromise? Or should I really keep it isolated to `parse_file` calls?

Reply:
We're pretty good at avoiding global variables elsewhere, and this seems like an easy case to handle. E.g. pass in a warning set/dict which the function adds to, which is nice in that it moves prints higher up the call stack too.